Tribler/community/multichain/community.py

```diff
@@ -376,12 +377,21 @@ def get_page_rank(self, public_key):
         :param public_key: the public key of the given node
         :return: the page rank of the given node
         """
+        if len(self.ranks) == 0:
+            self.ranks = self.get_page_ranks()
+        if public_key in self.ranks:
+            return self.ranks[public_key]
+        return 0
+
+    def get_page_ranks(self):
+        """
+        Return a dictionary of page ranks of all nodes.
+
+        :return: a dictionary of page ranks of all nodes
+        """
         self.page_rank.initial_walk()
         self.page_rank.count()
-        ranks = self.page_rank.get_ranks()
-        if public_key in ranks:
-            return ranks[public_key]
-        return 0
+        return self.page_rank.get_ranks()
```

Comment:
Good idea, that is more similar to how it was, the only thing I have to do is cache the returned list in order to prevent the method `get_ranks` from being called more than once